features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -160,6 +160,24 @@ void ff_memfree(void *p)
 }
 
 // Implementation of diskio functions (see ChaN/diskio.h)
+static WORD disk_get_sector_size(BYTE pdrv)
+{
+    WORD ssize = _ffs[pdrv]->get_erase_size();
+    if (ssize < 512) {
+        ssize = 512;
+    }
+
+    MBED_ASSERT(ssize >= _MIN_SS && ssize <= _MAX_SS);
```

Comment:
Alright, I forgot the fields are populated after the `init()`. I'd still move the assertions in the init codepath since those don't really change and the read/write paths should be as fast as possible.